ament_tools/build_types/bazel.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ament_tools.build_type import BuildAction
+from ament_tools.build_type import BuildType
+
+from ament_tools.context import ContextExtender
+
+from ament_tools.helper import extract_argument_group
+
+
+class BazelBuildType(BuildType):
+    build_type = 'bazel'
+    description = 'bazel project'
+
+    def prepare_arguments(self, parser):
+        parser.add_argument(
+            '--bazel-args',
+            nargs='*',
+            default=[],
+            help='Arbitrary arguments which are passed to all bazel projects. '
+            "Argument collection can be terminated with '--'.")
+
+    def argument_preprocessor(self, args):
+        args, bazel_args = extract_argument_group(args, '--bazel-args')
+        extras = {
+            'bazel_args': bazel_args,
+        }
+        return args, extras
+
+    def extend_context(self, options):
+        ce = ContextExtender()
+        ce.add('bazel_args', getattr(options, 'bazel_args', []))
+        return ce
+
+    def on_build(self, context):
+        yield BuildAction(['bazel', 'build'] + context.bazel_args + ['//...'],
+                          cwd=context.source_space)
```

Comment:
Please don't align argument with the parenthesis.

Same below.

Reply:
I've changed this to be a separate variable for the cmd instead.